neo4j/transaction_with_context.go

```diff
@@ -54,14 +54,32 @@ type ExplicitTransaction interface {
 	legacy() Transaction
 }
 
+type transactionState struct {
+	err                 error
+	resultErrorHandlers []func(error)
+}
+
+func newTransactionState() *transactionState {
+	return &transactionState{
+		err:                 nil,
+		resultErrorHandlers: make([]func(error), 0),
```

Comment:
More for my education: is this for all we're doing with it equivalent?
```suggestion
		resultErrorHandlers: nil,
```

Reply:
The `make` will allocate and return a non-nil slice of length and capacity 0 rather than having a nil value. However, in our case using nil will be fine as both `append` and iterate work on nil slices so this is now redundant. We can get rid of the newTransaction constructor. I'll have a refactor 👍